Crash Report 3.3.1: java.lang.IndexOutOfBoundsException CoreMainActivity.selectTab

```
Caused by: java.lang.IndexOutOfBoundsException: 

 
  at java.util.ArrayList.get (ArrayList.java:411)

 
  at org.kiwix.kiwixmobile.core.main.CoreMainActivity.selectTab (CoreMainActivity.java:872)

 
  at org.kiwix.kiwixmobile.core.main.CoreMainActivity.hideTabSwitcher (CoreMainActivity.java:572)

 
  at org.kiwix.kiwixmobile.core.main.CoreMainActivity.onActivityResult (CoreMainActivity.java:1479)
```

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java
@@ -868,8 +868,7 @@ public abstract class CoreMainActivity extends BaseActivity
   protected void selectTab(int position) {
     currentWebViewIndex = position;
     contentFrame.removeAllViews();
-
-    KiwixWebView webView = webViewList.get(position);
+    KiwixWebView webView = safelyGetWebView(position);
     if (webView.getParent() != null) {
       ((ViewGroup) webView.getParent()).removeView(webView);
     }
@@ -884,6 +883,16 @@ public abstract class CoreMainActivity extends BaseActivity
     if (!isHideToolbar && webView instanceof ToolbarScrollingKiwixWebView) {
       ((ToolbarScrollingKiwixWebView) webView).ensureToolbarDisplayed();
     }
+  }
+
+  private KiwixWebView safelyGetWebView(int position) {
+    return webViewList.size() == 0 ? newMainPageTab() : webViewList.get(safePosition(position));
+  }
+
+  private int safePosition(int position) {
+    return position < 0 ? 0
+      : position >= webViewList.size() ? webViewList.size() - 1
+        : position;
   }
 
   protected KiwixWebView getCurrentWebView() {

